pkg/net/url.go

```diff
@@ -144,6 +145,9 @@ func IsNetworkOrHostDown(err error) bool {
 	if err == nil {
 		return false
 	}
+	if errors.Is(err, context.Canceled) {
+		return false
+	}
```

Comment:
I think it would be reasonable to assume that if it is `context.DeadlineExceeded` then it should return true.

Reply:
That is automatically handled apparently because it's wrapped inside net.Error